src/docker/remote.rs

```diff
@@ -92,7 +92,9 @@ impl<'a, 'b, 'c> ContainerDataVolume<'a, 'b, 'c> {
             file::create_dir_all(dst_path.parent().expect("must have parent"))?;
             fs::copy(&src_path, &dst_path)?;
         }
-        self.copy_files(temppath, dst, msg_info)
+        // if copying from a src directory to dst directory with docker, to
+        // copy the contents from `src` into `dst`, `src` must end with `/.`
+        self.copy_files(&temppath.join("."), dst, msg_info)
```

Comment:
should this maybe be checked and done in `copy_files` ?

i.e the functions does it automagically for you

Reply:
I'll have to check: it's quite possible `copy_cargo` or `copy_rust` relies on this, but it would be trivial to change it to copy-in rather than just copy the entire directory. But this might also need patching elsewhere (looking at `copy_files_nocache`).

Reply:
So we can't do this entirely (see `rust_base`, where we want to copy into the folder), but we can do it for `copy_files_nocache`, so everything for the project files should be quite doable.

```rust
        let dst = mount_prefix.join(&dirs.sysroot_mount_path_relative()?);
        let rustlib = Path::new("lib").join("rustlib");
        self.create_dir(&dst.join(&rustlib), msg_info)?;
        for basename in ["bin", "libexec", "etc"] {
            let file = dirs.get_sysroot().join(basename);
            self.copy_files(&file, &dst, msg_info)?;
        }
```

See here, we want to copy `bin`, `libexec`, `etc` into `rustlib`, so to `rustlib/bin`, etc.